hack/dind-cluster.sh

```diff
@@ -210,27 +206,40 @@ function add-registry-to-runtime() {
-      ${DOCKER_CMD} exec -t ${cid} sh -c "sed -i '/^\[registries.insecure\]$/{\$!{N;s/^\[registries.insecure\]\nregistries .*/\[registries.insecure\]\nregistries = \[\"${registry}\"\]/g;ty;P;D;:y}}' /etc/containers/registries.conf"
-      ${DOCKER_CMD} exec -t ${cid} sh -c "systemctl restart docker"
+      error=0
+      ${DOCKER_CMD} exec -t ${cid} sh -c "grep -q ${registry} /etc/containers/registries.conf" || error=1
+      if [[ "${error}" -eq "1" ]]; then
+        ${DOCKER_CMD} exec -t ${cid} sh -c "sed -i '/^\[registries.search\]$/{\$!{N;s/^\[registries.search\]\nregistries .*/\[registries.search\]\nregistries = \[\"${registry}\", \"docker.io\", \"registry.fedoraproject.org\", \"registry.access.redhat.com\"\]/g;ty;P;D;:y}}' /etc/containers/registries.conf"
+        ${DOCKER_CMD} exec -t ${cid} sh -c "sed -i '/^\[registries.insecure\]$/{\$!{N;s/^\[registries.insecure\]\nregistries .*/\[registries.insecure\]\nregistries = \[\"${registry}\"\]/g;ty;P;D;:y}}' /etc/containers/registries.conf"
+        ${DOCKER_CMD} exec -t ${cid} sh -c "systemctl restart docker"
+      fi
     else
-      ${DOCKER_CMD} exec -t ${cid} sh -c "sed -i 's/^registries .*/registries = [\n\t\"${registry}\",/g' /etc/crio/crio.conf"
-      ${DOCKER_CMD} exec -t ${cid} sh -c "sed -i 's/^insecure_registries .*/insecure_registries = [\n\t\"${registry}\",/g' /etc/crio/crio.conf"
-      ${DOCKER_CMD} exec -t ${cid} sh -c "systemctl restart crio"
+      error=0
+      ${DOCKER_CMD} exec -t ${cid} sh -c "grep -q ${registry} /etc/crio/crio.conf" || error=1
+      if [[ "${error}" -eq "1" ]]; then
+        ${DOCKER_CMD} exec -t ${cid} sh -c "sed -i 's/^registries .*/registries = [\n\t\"${registry}\",/g' /etc/crio/crio.conf"
+        ${DOCKER_CMD} exec -t ${cid} sh -c "sed -i 's/^insecure_registries .*/insecure_registries = [\n\t\"${registry}\",/g' /etc/crio/crio.conf"
+        ${DOCKER_CMD} exec -t ${cid} sh -c "systemctl restart crio"
+      fi
     fi
   done
 }
 
-function enable-local-registry() {
+function ensure-local-registry() {
   local origin_root=$1
   local config_root=$2
   local ip=$3
   local images=($4)
   local tag=$(pushd "${origin_root}" >/dev/null ; git describe --abbrev=0 ; popd > /dev/null)
 
-  # Start local registry
-  ${DOCKER_CMD} run -d -p ${REGISTRY_PORT}:5000 --restart=always --name ${REGISTRY_NAME} registry:2 > /dev/null
-  echo "Created local registry '${REGISTRY_NAME}' at ${ip}:${REGISTRY_PORT}"
+  local cid=$( ${DOCKER_CMD} ps -qa --filter "name=${REGISTRY_NAME}" )
+  if [[ -z "${cid}" ]]; then
+    # Start local registry
+    ${DOCKER_CMD} run -d -p ${REGISTRY_PORT}:5000 --restart=always --name ${REGISTRY_NAME} registry:2 > /dev/null
+    echo "  Created local registry '${REGISTRY_NAME}' at ${ip}:${REGISTRY_PORT}"
+  else
+    echo "  Found local registry '${REGISTRY_NAME}' at ${ip}:${REGISTRY_PORT}"
```

Comment:
This could lead to confusion if the registry already exists but with a different IP address or port.